lib/mayaUsd/resources/ae/usd-shared-components/src/python/usdSharedComponents/common/filteredStringListView.py

```diff
@@ -65,101 +67,76 @@ def paint(
         painter = QPainter(self)
         painter.fillRect(event.rect(), Theme.instance().palette.colorListBorder)
         super(FilteredStringListView, self).paintEvent(event)
-        if self._model.isFilteredEmpty():
-            painter = QPainter(self.viewport())
-            painter.setPen(Theme.instance().palette.colorPlaceHolderText)
-            painter.drawText(self.rect(), Qt.AlignCenter, kNoObjectFoundLabel)
-
-    @property
-    def items(self) -> Sequence[str]:
-        return self._model.stringList
-
-    @items.setter
-    def items(self, items: Sequence[str]):
-        # save the selected items
-        prevSelectedIndices = []
-        for s in self.selectedItems:
-            try:
-                newIdx = items.index(s)
-            except ValueError:
-                continue
-            prevSelectedIndices.append(newIdx)
-
-        self._model.setStringList(items)
-        # restore the selected items
-        newSelection = QItemSelection()
-        for i in prevSelectedIndices:
-            idx = self._model.index(i, 0)
-            newSelection.select(idx, idx)
-        self.selectionModel().select(newSelection, QItemSelectionModel.ClearAndSelect)
-        self.updatePlaceholder()
-
-    @property
+
+        model = self.model()
+        if not model:
+            return
+
+        if model.isFilteredEmpty():
+            self._paintPlaceHolder(NO_OBJECTS_FOUND_LABEL)
+        elif model.rowCount() == 0:
+            if Host.instance().canDrop and Host.instance().canPick:
+                self._paintPlaceHolder(DRAG_OR_PICK_OBJECTS_LABEL)
+            elif Host.instance().canDrop:
+                self._paintPlaceHolder(DRAG_OBJECTS_HERE_LABEL)
+            elif Host.instance().canPick:
+                self._paintPlaceHolder(PICK_OBJECTS_LABEL)
+
     def selectedItems(self) -> Sequence[str]:
         return [str(index.data(Qt.DisplayRole)) for index in self.selectedIndexes()]
 
-    @property
     def hasSelectedItems(self) -> bool:
-        return self.selectionModel().hasSelection()
-
-    def updatePlaceholder(self):
-        """Show or hide placeholder based on the model's content."""
-        if self.placeholder_label is not None:
-            model = self.model()
-            if model and model.rowCount() == 0:
-                self.placeholder_label.show()
-            else:
-                self.placeholder_label.hide()
-
-    def resizeEvent(self, event):
-        """Ensure placeholder is centered on resize."""
-        super().resizeEvent(event)
-        if self.placeholder_label is not None:
-            self.placeholder_label.setGeometry(self.viewport().geometry())
+        return bool(self.selectionModel().hasSelection())
+
+    def _paintPlaceHolder(self, placeHolderText):
+        painter = QPainter(self.viewport())
+        theme = Theme.instance()
+        painter.setPen(theme.palette.colorPlaceHolderText)
+        painter.drawText(self.rect(), Qt.AlignCenter, placeHolderText)
```

Comment:
could be moved inside the paintEvent() override - to avoid being accidentally called outside the paintEvent - what must not be done in Qt.

Reply:
True even if unlikely. I will move it in a subsequent PR that is based on this one. (Not another refactor one :) )